From f02412bb406ea2e42c7e5354e5ca0ddd412b71c6 Mon Sep 17 00:00:00 2001
From: Radha Mohan Chintakuntla <radhac@marvell.com>
Date: Tue, 28 Aug 2018 13:41:53 -0700
Subject: [PATCH 0172/1051] gpio: thunderx: Add workaround for errata 34800

This patch implements workaround for errata 34800 applicable to CN96XX
SoCs of Marvell.

Signed-off-by: Radha Mohan Chintakuntla <radhac@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/gpio/gpio-thunderx.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpio/gpio-thunderx.c b/drivers/gpio/gpio-thunderx.c
index b82d214bd420..de098ec22f8e 100644
--- a/drivers/gpio/gpio-thunderx.c
+++ b/drivers/gpio/gpio-thunderx.c
@@ -16,7 +16,6 @@
 #include <linux/pci.h>
 #include <linux/spinlock.h>
 
-
 #define GPIO_RX_DAT	0x0
 #define GPIO_TX_SET	0x8
 #define GPIO_TX_CLR	0x10
@@ -505,7 +504,14 @@ static int thunderx_gpio_probe(struct pci_dev *pdev,
 		u64 c = readq(txgpio->register_base + GPIO_CONST);
 
 		ngpio = c & GPIO_CONST_GPIOS_MASK;
-		txgpio->base_msi = (c >> 8) & 0xff;
+
+		/* Workaround for Errata 34800 */
+		if (MIDR_IS_CPU_MODEL_RANGE(read_cpuid_id(),
+		    MIDR_MRVL_OCTEONTX2_96XX, 0x00, 0xf)) {
+			txgpio->base_msi = 0x36;
+		} else {
+			txgpio->base_msi = (c >> 8) & 0xff;
+		}
 	}
 
 	txgpio->msix_entries = devm_kcalloc(dev,
-- 
2.17.1

